app/lib/decorators.py

```diff
@@ -116,3 +118,43 @@ def decorated_function(*args, **kwargs):
         return f(*args, **kwargs)
 
     return decorated_function
+
+
+class skip_and_return_if(object):
+    """A decorator that will skip the wrapped function and automatically return
+    ``None`` if the passed ``flag`` is ``True``.
+
+    Decorators like this that take parameters themselves must be
+    implemented either as a three nested functions (ugly), or as a class
+    (this implementation).  The ``__init__`` method creates the decorator,
+    and is passed any arguments for how the decorator should behaive, and
+    then the ``__call__`` method
+    """
+
+    def __init__(self, flag):
+        """Create a decorator to skip the decorated function and return
+        ``None`` if ``flag`` is ``True``.
+
+        :param bool flag: Skip the decorated function if this is ``True``.
+        """
+        self.flag = flag
+
```

Comment:
2 between functions at the module level, 1 between functions within classes.

(preparing a pull request for applying this standard to all of our python files.  It's properly applied here though)

https://www.python.org/dev/peps/pep-0008/#blank-lines